[BUG] Issue with MIND test

### Description
<!--- Describe your issue/bug/request in detail -->
Error https://github.com/recommenders-team/recommenders/actions/runs/21589651860/job/62206326304
```
tests/data_validation/recommenders/datasets/test_movielens.py .......... [ 29%]
..                                                                       [ 35%]
tests/data_validation/recommenders/datasets/test_mind.py FFFFFFFFF...... [ 79%]
                                                                         [ 79%]
tests/data_validation/examples/test_mind.py ..                           [ 85%]
tests/data_validation/examples/test_wikidata.py ss                       [ 91%]
tests/smoke/examples/test_notebooks_python.py ..                         [ 97%]
tests/functional/examples/test_notebooks_python.py .                     [100%]

pytest tests/data_validation/recommenders/datasets/test_mind.py

    @pytest.mark.parametrize(
        "url, content_length, etag",
        [
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_train.zip",
                "17372879",
                '"0x8D8B8AD5B233930"',
            ),  # NOTE: the z20 blob returns the etag with ""
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_dev.zip",
                "10080022",
                '"0x8D8B8AD5B188839"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_utils.zip",
                "97292694",
                '"0x8D8B8AD5B126C3B"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_train.zip",
                "52994575",
                '"0x8DCC5A830190676"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_dev.zip",
                "30948560",
                '"0x8DCC5A82E182A0F"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_utils.zip",
                "155178106",
                '"0x8D8B8AD5B3677C6"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_train.zip",
                "531360717",
                '"0x8DCC5A8375BDC1D"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_dev.zip",
                "103592887",
                '"0x8DCC5A82FE8609C"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_utils.zip",
                "150359301",
                '"0x8D8B8AD5B2ED4C9"',
            ),
        ],
    )
    def test_mind_url(url, content_length, etag):
        url_headers = requests.head(url).headers
>       assert url_headers["Content-Length"] == content_length
E       AssertionError: assert '1363' == '17372879'
E         
E         - 17372879
E         + 1363

tests/data_validation/recommenders/datasets/test_mind.py:63: AssertionError
_ test_mind_url[https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_dev.zip-10080022-"0x8D8B8AD5B188839"] _

url = 'https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_dev.zip'
content_length = '10080022', etag = '"0x8D8B8AD5B188839"'

    @pytest.mark.parametrize(
        "url, content_length, etag",
        [
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_train.zip",
                "17372879",
                '"0x8D8B8AD5B233930"',
            ),  # NOTE: the z20 blob returns the etag with ""
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_dev.zip",
                "10080022",
                '"0x8D8B8AD5B188839"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_utils.zip",
                "97292694",
                '"0x8D8B8AD5B126C3B"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_train.zip",
                "52994575",
                '"0x8DCC5A830190676"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_dev.zip",
                "30948560",
                '"0x8DCC5A82E182A0F"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_utils.zip",
                "155178106",
                '"0x8D8B8AD5B3677C6"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_train.zip",
                "531360717",
                '"0x8DCC5A8375BDC1D"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_dev.zip",
                "103592887",
                '"0x8DCC5A82FE8609C"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_utils.zip",
                "150359301",
                '"0x8D8B8AD5B2ED4C9"',
            ),
        ],
    )
    def test_mind_url(url, content_length, etag):
        url_headers = requests.head(url).headers
>       assert url_headers["Content-Length"] == content_length
E       AssertionError: assert '1363' == '10080022'
E         
E         - 10080022
E         + 1363

tests/data_validation/recommenders/datasets/test_mind.py:63: AssertionError
_ test_mind_url[https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_utils.zip-97292694-"0x8D8B8AD5B126C3B"] _

url = 'https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_utils.zip'
content_length = '97292694', etag = '"0x8D8B8AD5B126C3B"'

    @pytest.mark.parametrize(
        "url, content_length, etag",
        [
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_train.zip",
                "17372879",
                '"0x8D8B8AD5B233930"',
            ),  # NOTE: the z20 blob returns the etag with ""
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_dev.zip",
                "10080022",
                '"0x8D8B8AD5B188839"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_utils.zip",
                "97292694",
                '"0x8D8B8AD5B126C3B"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_train.zip",
                "52994575",
                '"0x8DCC5A830190676"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_dev.zip",
                "30948560",
                '"0x8DCC5A82E182A0F"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_utils.zip",
                "155178106",
                '"0x8D8B8AD5B3677C6"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_train.zip",
                "531360717",
                '"0x8DCC5A8375BDC1D"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_dev.zip",
                "103592887",
                '"0x8DCC5A82FE8609C"',
            ),
            (
                "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_utils.zip",
                "150359301",
                '"0x8D8B8AD5B2ED4C9"',
            ),
        ],
    )
    def test_mind_url(url, content_length, etag):
        url_headers = requests.head(url).headers
>       assert url_headers["Content-Length"] == content_length
E       AssertionError: assert '1367' == '97292694'
E         
E         - 97292694
E         + 1367
```


### In which platform does it happen?
<!--- Describe the platform where the issue is happening (use a list if needed) -->
<!--- For example: -->
<!--- * Azure Data Science Virtual Machine. -->
<!--- * Azure Databricks.  -->
<!--- * Other platforms.  -->

### How do we replicate the issue?
<!--- Please be specific as possible (use a list if needed). -->
<!--- For example: -->
<!--- * Create a conda environment for pyspark -->
<!--- * Run unit test `test_sar_pyspark.py` with `pytest -m 'spark'` -->
<!--- * ... -->

### Expected behavior (i.e. solution)
<!--- For example:  -->
<!--- * The tests for SAR PySpark should pass successfully. -->

### Willingness to contribute
<!--- Go over all the following points, and put an `x` in the box that apply. -->
- [ ] Yes, I can contribute for this issue independently.
- [ ] Yes, I can contribute for this issue with guidance from Recommenders community.
- [ ] No, I cannot contribute at this time.

### Other Comments

--- a/tests/data_validation/recommenders/datasets/test_mind.py
+++ b/tests/data_validation/recommenders/datasets/test_mind.py
@@ -14,52 +14,52 @@ from recommenders.datasets.mind import download_mind, extract_mind
         (
             "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_train.zip",
             "17372879",
-            '"0x8D8B8AD5B233930"',
-        ),  # NOTE: the z20 blob returns the etag with ""
+            '"66b4bd16ce4322e71ed7f2ec238cf0708d6b0b04651fe7e9bae7b7dd5e17bc40"',
+        ),
         (
             "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_dev.zip",
             "10080022",
-            '"0x8D8B8AD5B188839"',
+            '"99def74b59c7e423854b79d4b68757e9e49813d8fc0cce6ee172bf5b3240f01d"',
         ),
         (
             "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDdemo_utils.zip",
             "97292694",
-            '"0x8D8B8AD5B126C3B"',
+            '"0b887006ffc2db8572366ae7d529c9eee24f0eb915c75ffc0963437655b10711"',
         ),
         (
             "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_train.zip",
             "52994575",
-            '"0x8DCC5A830190676"',
+            '"0d6d50cfb534b494a1153694f8a488170b28b0ac1c254363b565aa6b11c7761f"',
         ),
         (
             "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_dev.zip",
             "30948560",
-            '"0x8DCC5A82E182A0F"',
+            '"e3db712b7eb4339bc1729c5a0075c2e8204107690f9d5f0f974de4852622564f"',
         ),
         (
             "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDsmall_utils.zip",
             "155178106",
-            '"0x8D8B8AD5B3677C6"',
+            '"8047c5282f4ee819389e6ad757841e93705a7988af7d851184047c05337114ed"',
         ),
         (
             "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_train.zip",
             "531360717",
-            '"0x8DCC5A8375BDC1D"',
+            '"6a73f3b9ab3ba208895a18c1ac542dcf6bc9c7b817af433e60282d041cfec427"',
         ),
         (
             "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_dev.zip",
             "103592887",
-            '"0x8DCC5A82FE8609C"',
+            '"526a6486eed6b4053b9460c42aac7239c1234cfecaf82a3c345b742d3521642f"',
         ),
         (
             "https://huggingface.co/datasets/Recommenders/MIND/resolve/main/MINDlarge_utils.zip",
             "150359301",
-            '"0x8D8B8AD5B2ED4C9"',
+            '"8e05c4c768336dd8af8bd05d098a3572f657c984a5415e64fc85aaa315a32e15"',
         ),
     ],
 )
 def test_mind_url(url, content_length, etag):
-    url_headers = requests.head(url).headers
+    url_headers = requests.head(url, allow_redirects=True).headers
     assert url_headers["Content-Length"] == content_length
     assert url_headers["ETag"] == etag
 

